api/app/models/job_file.rb

```diff
@@ -32,30 +32,38 @@ class JobFile
   class << self
     prepend FlightJobScriptAPI::ModelCache
 
-    def index_job_results!(job_id, user:)
-      # Load up the job
-      job = Job.find!(job_id, user: user)
-      return nil unless job
-      return nil unless job.metadata['results_dir']
-      results_dir = job.metadata['results_dir']
-
-      # Find the relevant files
-      cmd = FlightJobScriptAPI::SystemCommand.recursive_glob_dir(results_dir, user: user)
-
-      # Return empty array if the results directory is missing
-      return [] if cmd.exitstatus == 20
+    def generate_file_id(results_dir, file_path)
+      Base64.urlsafe_encode64 Pathname.new(file_path).relative_path_from(results_dir).to_s
+    end
 
-      # Error if the system command otherwise failed
-      unless cmd.exitstatus == 0
-        raise CommandError, "Failed to load the result files for job: #{job_id}"
+    # NOTE: There is a bit of back and forth between Job and JobFile here
+    #       This is due to how the 'includes' was retrofitted over the original implementation
+    #       Consider refactoring
+    def index_job_results!(job_id, user:)
+      # Attempt to load the cached version of Job
+      job = Job.find!(job_id, user: user, include: ['result_files'])
+
+      # NOTE: If the Job is pre-cached, it must be loaded with --include result_files
+      # This *should* already be the case due to the 'includes' mechanism
+      # Revisit if required
+      unless job.metadata.key? 'result_files'
+        FlightJobScriptAPI.logger.error <<~ERROR.chomp
+          The job '#{job_id}' was unexpectedly loaded without its results_files
+        ERROR
+        raise FlightJobScriptAPI::CommandError, "Failed to load the results files for job: #{job_id}"
```

Comment:
It is a difficult one to recover from because the recursive globing has been removed.

In practice it would only happen if a dev makes a mistake and misses an `includes`.

Reply:
So `job` is loaded on line 44.  It is either retrieved from the cache, or is newly loaded.  If newly loaded, it will have the appropriate `include`, because it is right there on line 44.  If it is retrieved from the cache, it may or may not have the include.  You're claim is that it is a programmer error if it doesn't.

The route `GET /job/:id/result-files` appears to not include the required `include` for the job.

Reply:
On review I agree this is too complicated. I've removed the conditional check entirely:

https://github.com/openflighthpc/flight-job-script-service/pull/38/commits/282b56664315a41ddc2f8ad59624b80b649958a0

Instead I am going to fix the issue in the CLI

Reply:
https://github.com/openflighthpc/flight-job/pull/26/commits/976be840d0df16231d241d62fb6e7024caa50c46